packages/strapi/lib/commands/generate-template.js

```diff
@@ -16,13 +16,17 @@ const TEMPLATE_CONTENT = ['api', 'components', 'config/functions/bootstrap.js',
 async function copyContent(templatePath, rootBase) {
   for (const item of TEMPLATE_CONTENT) {
     try {
-      await fse.copy(join(process.cwd(), item), join(templatePath, item));
-      const currentProjectBase = basename(process.cwd());
-      console.log(
-        `${chalk.green(
-          'success'
-        )}: copy ${currentProjectBase}/${item} => ${rootBase}/template/${item}`
-      );
+      const folderExists = await fse.pathExists(join(process.cwd(), item));
+
+      if (folderExists) {
+        await fse.copy(join(process.cwd(), item), join(templatePath, item));
+        const currentProjectBase = basename(process.cwd());
+        console.log(
+          `${chalk.green(
+            'success'
+          )}: copy ${currentProjectBase}/${item} => ${rootBase}/template/${item}`
+        );
+      }
```

Comment:
```suggestion
      const pathToCopy = join(process.cwd(), item);
      
      if(!await fse.pathExists(pathToCopy)) { 
        return
      };

      await fse.copy(pathToCopy, join(templatePath, item));
      const currentProjectBase = basename(process.cwd());
      console.log(
        `${chalk.green(
          'success'
        )}: copy ${currentProjectBase}/${item} => ${rootBase}/template/${item}`
      );
```

Reply:
But if one directory doesn't exist won't returning early mean the rest of the folders are ignored?

A project may not have `components`, `data`, or `bootstrap.js`  but the template should still be able to be created, as long as at least an `api` folder exists.

Reply:
You must use continue in a for loop :)